flepimop/gempyor_pkg/src/gempyor/distributions.py

```diff
@@ -0,0 +1,373 @@
+"""Representations of distributions used for modifiers, likelihoods, etc."""
+
+__all__: tuple[str, ...] = (
+    "BinomialDistribution",
+    "Distribution",
+    "DistributionABC",
+    "FixedDistribution",
+    "GammaDistribution",
+    "LognormalDistribution",
+    "NormalDistribution",
+    "PoissonDistribution",
+    "TruncatedNormalDistribution",
+    "UniformDistribution",
+    "WeibullDistribution",
+)
+
+
+from abc import ABC, abstractmethod
+from math import isclose
+from typing import Annotated, Literal
+
+import numpy as np
+from numpy.random import Generator
+import numpy.typing as npt
+from pydantic import BaseModel, Field, model_validator
+from scipy.stats import truncnorm
+
+
+class DistributionABC(ABC, BaseModel):
+    """Base class for distributions used in modifiers, likelihoods, etc."""
+
+    distribution: str
+
+    @abstractmethod
+    def sample(
+        self, size: int | tuple[int, ...] = 1, rng: Generator | None = None
+    ) -> npt.NDArray[np.float64 | np.int64]:
+        """
+        Draw random sample(s) from the distribution.
+
+        Args:
+            size: The desired output size of samples to be drawn.
+            rng: A NumPy random number generator instance used for sampling to ensure reproducibility.
+        """
```

Comment:
```suggestion

        Returns:
            A numpy array of either floats/ints (depending on distribution)
            with shape `size` drawn from the distribution.
        """
```
Or similar for a complete docstring.